openlr/openlr_tests/decoded_path_test.cpp

```diff
@@ -0,0 +1,194 @@
+#include "testing/testing.hpp"
+
+#include "openlr/decoded_path.hpp"
+
+#include "generator/generator_tests_support/test_feature.hpp"
+#include "generator/generator_tests_support/test_mwm_builder.hpp"
+
+#include "indexer/classificator_loader.hpp"
+#include "indexer/index.hpp"
+
+#include "coding/file_name_utils.hpp"
+
+#include "platform/country_file.hpp"
+#include "platform/local_country_file.hpp"
+#include "platform/platform.hpp"
+#include "platform/platform_tests_support/scoped_dir.hpp"
+#include "platform/platform_tests_support/scoped_file.hpp"
+
+#include <algorithm>
+#include <iomanip>
+#include <sstream>
+#include <vector>
+
+#include "3party/pugixml/src/goodies.hpp"
+#include "3party/pugixml/src/pugixml.hpp"
+
+using namespace platform;
+using namespace platform::tests_support;
+using namespace generator::tests_support;
+
+namespace
+{
+string const kTestDir = "openlr_decoded_path_test";
+string const kTestMwm = "test";
+
+double Ceil(double d)
+{
+  std::stringstream s;
+  s << std::setprecision(5) << std::fixed;
+  s << d;
+  s >> d;
+  return d;
+}
+
+m2::PointD CeilPoint(m2::PointD const & p)
+{
+  return {Ceil(p.x), Ceil(p.y)};
+}
+
+routing::Junction CeilJunction(routing::Junction const & j)
+{
+  return routing::Junction(
+      CeilPoint(j.GetPoint()),
+      j.GetAltitude()
+  );
+}
+
+routing::Edge CeilEdgeJunctions(routing::Edge const & e)
+{
+  return routing::Edge(
+      e.GetFeatureId(),
+      e.IsForward(),
+      e.GetSegId(),
+      CeilJunction(e.GetStartJunction()),
+      CeilJunction(e.GetEndJunction())
+  );
+}
+
+void CeilJunctionsInPath(openlr::Path & p)
+{
+  for (auto & e : p)
+    e = CeilEdgeJunctions(e);
+}
+
+void TestSerializeDeserialize(openlr::Path path, Index const & index)
+{
+  pugi::xml_document doc;
+  openlr::PathToXML(path, doc);
+
+  openlr::Path restoredPath;
+  openlr::PathFromXML(doc, index, restoredPath);
+
+  // Fix MercatorBounds::From/ToLatLon floating point error
+  // for we could use TEST_EQUAL on result.
+  CeilJunctionsInPath(path);
+  CeilJunctionsInPath(restoredPath);
+
+  TEST_EQUAL(path, restoredPath, ());
+}
+
+openlr::Path MakePath(FeatureType const & road, bool const forward)
+{
+  CHECK_EQUAL(road.GetFeatureType(), feature::GEOM_LINE, ());
+  openlr::Path path;
+
+  if (forward)
```

Comment:
Please move the "if" inside. You only really need to split the calculations of ```from``` and ```to```.